pkg/logs/input/windowsevent/tailer.go

```diff
@@ -68,10 +66,150 @@ func (t *Tailer) toMessage(event string) (*message.Message, error) {
 	if err != nil {
 		return &message.Message{}, err
 	}
+	if strings.HasPrefix(t.config.ChannelPath, "Microsoft-ServiceFabric/") {
+		mapTaskIDToTaskName(mv)
+	}
 	jsonEvent, err := mv.Json(false)
 	if err != nil {
 		return &message.Message{}, err
 	}
 	log.Debug("Sending JSON: ", string(jsonEvent))
 	return message.NewMessage(jsonEvent, message.NewOrigin(t.source), message.StatusInfo), nil
 }
+
+// mapTaskIDToTaskName looks for the TASK_ID in {"Event": {"System": {"Task": <TASK_ID> }}}
+// and maps it to the name of the task that match in the Microsoft Task Codes
+func mapTaskIDToTaskName(mv mxj.Map) {
+	if evt, ok := mv["Event"]; ok {
+		if sys, ok := evt.(map[string]interface{})["System"]; ok {
+			if taskID, ok := sys.(map[string]interface{})["Task"]; ok {
+				if taskName, ok := taskIDMapping[taskID]; ok {
+					sys.(map[string]interface{})["Task"] = taskName
+				}
+			}
+		}
+	}
+}
+
+// Mapping can be found here
+// https://github.com/Microsoft/service-fabric/blob/c326b801c6c709f36684700edfe7bb88ceec9d7f/src/prod/src/Common/TraceTaskCodes.h
+var taskIDMapping = map[interface{}]string{
```

Comment:
map[string]string ?

Reply:
That would force us to do some type casting in the calling method. I prefer this because it avoids one error checking (has the type cast succeeded)